tests/unit-tests/api/payment-gateways.php

```diff
@@ -269,6 +269,69 @@ public function test_payment_gateway_schema() {
 		$this->assertArrayHasKey( 'settings', $properties );
 	}
 
+	/**
+	 * Test system_status _filter query parameter.
+	 */
+	public function test_get_payment_gateways_info_filtered() {
+		wp_set_current_user( $this->user );
+
+		$query_params = array(
+			'_fields' => 'id,title,nonexisting',
+		);
+		$request      = new WP_REST_Request( 'GET', '/wc/v2/payment_gateways' );
+		$request->set_query_params( $query_params );
+
+		$response = $this->server->dispatch( $request );
+		$data     = $response->get_data();
+		$this->assertEquals( 200, $response->get_status() );
+
+		$one_gateway_info = reset( $data );
+		$this->assertEquals( 2, count( $one_gateway_info ) );
+
+		// Selected fields returned in the response.
+		$this->assertArrayHasKey( 'id', $one_gateway_info );
+		$this->assertArrayHasKey( 'title', $one_gateway_info );
+
+		// Fields not selected omitted from response.
+		$this->assertArrayNotHasKey( 'order', $one_gateway_info );
```

Comment:
@peterfabian`assertArrayNotHasKey` is easier and more consistent than `$this->assertTrue( ! isset( $item['foo'] ) );` Just a heads-up for future tests. :)

Reply:
True that, and  less error prone, as I've noticed that `! isset( $this_does_not_exist_or_just_mispelled_var )` is true even though it might not be really what the tester intended. I updated tests in my PRs, too.

Reply:
One thing to mention here is, though, that we should be merging these to `rest-api-v3` branch (as I found out yesterday while discussing with our API magician Claudio).